api/proxy/metrics/metrics.go

```diff
@@ -57,12 +57,14 @@ type Metrics struct {
 
 var _ Metricer = (*Metrics)(nil)
 
-func NewMetrics(subsystem string) *Metrics {
+func NewMetrics(registry *prometheus.Registry, subsystem string) *Metrics {
```

Comment:
is there any reason to keep the "subsystem" around? This would be useful if we reused this struct for different subsystems (as for logs for example), but afaiu this metrics struct is only ever created once, for the proxy, and we just use "default" subsystem. No need in having it as an argument imo

Reply:
Removed subsystem arg in: [e2f81e3](https://github.com/Layr-Labs/eigenda/pull/1863/commits/e2f81e3e1d1986a53d5f4439a28c93f3febc7b67)